pontoon/base/tests/test_views.py

```diff
@@ -140,3 +143,39 @@ def test_no_subpage_no_stats_in_current_locale(self):
         with patch('pontoon.base.views.render', wraps=render) as mock_render:
             self.client.get(url)
             assert_true('part' not in mock_render.call_args[0][2])
+
+
+class ContributorsTests(TestCase):
```

Comment:
The two tests you've added only test that the queryset within the view has the same length as the number of users created by your test, which doesn't really test the functionality we want.

These are the tests I'd expect for a pull request like this:

- View tests
  - Test that when `period` is not specified, that `start_date` (when passed to `with_translation_counts`) is `None` and `period` in the template context is `None`. You'll want to mock out `with_translation_counts` so you can inspect the argument passed to it.
  - Similar test to the one above, but for the case when `period` is not a valid `int`.
  - Test that when period is a valid int, `start_date` is calculated correctly based on the current date. You'll want to use `patch` to mock out the `timezone.now` function in the view so that the test doesn't rely on the current date when it's running. See [here](https://github.com/mozilla/pontoon/blob/master/pontoon/administration/tests/test_sync_projects.py#L41) for an example of a test that does this.
- `UserTranslationsManagerTests`
  - Test that users who have an email in `settings.EXCLUDE` aren't included in the results from `with_translation_counts`.
  - Test that users who have no translations whatsoever aren't included in the results from `with_translation_counts`.
  - Test that the number of users returned by `with_translation_counts` is limited to the `limit` parameter.
  - Test that users returned by `with_translation_counts` are ordered by their translation count.
  - Test that the 4 attributes added by `with_translation_counts` are calculated correctly. For this one you'll want to create 1 or 2 users and some translations with various combinations of `fuzzy` and `approved`, and then assert that the stats that are added by `with_translation_counts` matches the test data you created.
  - Test that the stats added by `with_translation_counts` do not count translations that are identical to their source value. This could possibly be combined with the previous test if you also create some identical translations before checking the stats.

It's a long list but this code does a lot of stuff. :P